### PR TITLE
Fix DRY in Readmes and small errors

### LINKED ISSUE #687: Large DRY violations in quickstart READMEs
**Summary**

We have page after page of repeated code, at the beginning of each README. This is going to be a maintenance nightmare. We've got to fix this.

**Background**

- DRY = Don't Repeat Yourself.
- Large DRY violation = big copy-and-paste of code, which is a 
problem because you have to maintain all of them. The ideal to aim for is "never have duplicates".

In the beginning of V3, we had 2 quickstart READMEs (simple flow, marketplace flow). They partly differed in which components needed installing. So for user simplicity, we made each of those quickstart READMEs be self-contained, even if there was slight overlap. Only a small DRY violation. It was maintainable because it was just two READMEs.

Then the C2D README got added. It also had slightly different component needs, so we let it be self-contained. Small DRY violation. It made things slightly harder to maintain, but not a huge problem.

But now in V4, we have three more quickstart-like READMEs, [see here](https://github.com/oceanprotocol/ocean.py/tree/v4main/READMEs): 
- dispenser-flow.md
- fixed-rate-exchange-flow.md
- get-test-ocean.md

In addition to the prior ones:
- datatokens-flow.md
- marketplace-flow.md
- c2d-flow.md

All of these READMEs are repeating the following sections - about a page of code at the top of each README :
- "Prerequisites" section (3 lines)
- "Run barge services"  (7 lines)
- "Install the library" section" (6 lines)

Also, all the READMEs have:
- "Set envvars" a bit, but several differences here (10 lines)

And, some of these READMEs have:
- "Alice publishes data asset" (or very similar) (35 lines)

Maintaining this is going to be a nightmare. More likely, it won't be well maintained, then we will be living with buggy READMEs that are always half-broken. Therefore we need to do this as part of V4, to maximize the chance of all ocean.py READMEs working in V4.

**TODOs**

Solve this, according to the goals:
- It's not a maintenance nightmare
- It's still a nice UX for users going through

Proposed approach: 
- The simplest quickstart (datatokens-flow.md) stays unchanged
- In other quickstarts:
  - At the very top, ask the user to go through the simplest quickstart first" and link to it. (datatokens-flow.md)
  - Deprecate: "pre-requisites", "run barge", "install library" sections 
  - Where possible, deprecate "envvars" section. 
  - Where possible, deprecate "publish data asset" section

### PR DESCRIPTION
**Pull request by md0x**
Signed-off-by: Pablo Maldonado <pablo@oceanprotocol.com>

Work on #687 .

Changes proposed in this PR:

- Fix DRY in readmes
- Fix dispenser test

Note: c2c-flow.md has also been updated but its outdated until https://github.com/oceanprotocol/ocean.py/pull/674 is merged

### CODE REVIEW

File: READMEs/datatokens-flow.md [hunk 1]
```diff
@@ -25,11 +26,14 @@ cd barge
 # Clean up old containers (to be sure)
 docker system prune -a --volumes
 
-# Run barge: start ganache, Provider, Aquarius; deploy contracts; update ~/.ocean
-./start_ocean.sh
+# make sure to use the dev version of the operator service URL, to match the barge provider
+export OPERATOR_SERVICE_URL=https://c2d-dev.operator.oceanprotocol.com/
+
+#run barge: start ganache, Provider, Aquarius; deploy contracts; update ~/.ocean
+./start_ocean.sh  --with-provider2
```
Comment (md0x) [inline READMEs/datatokens-flow.md:33]: I have added 
`export OPERATOR_SERVICE_URL=https://c2d-dev.operator.oceanprotocol.com/` as it was used in the rest of quickstarts and doesn't affect quickstart

The same for  adding `--with-provider2`
`./start_ocean.sh --with-provider2`

**Comment by trentmc** [comment]
This is a great start!

There's still work to do. From the original issue:
---
Also, all the READMEs have:
- "Set envvars" a bit, but several differences here (10 lines)

And, some of these READMEs have:
- "Alice publishes data asset" (or very similar) (35 lines)

**Comment by calina-c** [comment]
@trentmc can we merge partial progress? Since Maria and David are working on readmes right now, we could help out with what we can right now. And make it even better along the way.

**Comment by trentmc** [comment]
> @trentmc can we merge partial progress?

OK!

**Review by trentmc — APPROVED** [review]
Approving, with the assumption there will be follow-up PRs for the rest.

**Pull request merged**